### PR TITLE
FreeBSD, MSYS2, and Arch ports

### PR DESCRIPTION
**Pull request by DinCahill**
- FreeBSD `Makefile`, and MSYS2 and Arch `PKGBUILD`s.
- Actually add `make install`. 
- Fix adding `CFLAGS`.
- Remove playd_tests from ALL_BUILD. Now `make` doesn't build the tests. `make check` builds and runs the tests, and `make test` just runs them.

The `PKGBUILD`s build from git master. When you run `makepkg`, it will overwrite the placeholder `pkgver=git` to the correct version from `pkgver()`.

The FreeBSD port can only build from git tags. I've set it to `0.3.1`, to encourage us to release that soon :^)

Debian/Ubuntu packages would be nice if they weren't so disgusting to make. At least `checkinstall` works fine.

Ready 4 Review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,6 +270,7 @@ with their licence information attached.
 [libsox]:                 http://sox.sourceforge.net
 [libuv]:                  https://github.com/joyent/libuv
 [MIT licence]:            http://opensource.org/licenses/MIT
+[MSYS2]:		  https://msys2.github.io/
```
Comment (LordAro) [inline README.md]: Muh alignment

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Makefile
+./Makefile
```
Comment (LordAro) [inline .gitignore]: Why's the ./ necessary?

File: ports/arch/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+             "sdl2")
+depends=("libuv"
+         "mpg123"
+         "libsndfile"
+         "sdl2")
+
+provides=("ury-playd")
+conflicts=("ury-playd")
+
+source=("ury-playd-git::git+http://github.com/UniversityRadioYork/ury-playd.git#branch=master")
```
Comment (LordAro) [inline ports/arch/PKGBUILD]: fairly sure specifying the branch isn't necessary if it's master

File: ports/freebsd/Makefile [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Created by: Donal Cahill <din@ury.york.ac.uk>
+# $FreeBSD$
+
+PORTNAME=		ury-playd
+DISTVERSIONPREFIX=	v
+DISTVERSION=		0.3.1
+CATEGORIES=		audio
+
+MAINTAINER=	din@ury.york.ac.uk
```
Comment (LordAro) [inline ports/freebsd/Makefile]: *ury.org.uk

File: ports/freebsd/Makefile [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		libuv.so:devel/libuv
+BUILD_DEPENDS=	cmake:devel/cmake
+
+USES=	cmake compiler:c++14-lang
+
+PLIST_FILES=	bin/playd man/man1/playd.1.gz
+
+USE_GITHUB=	yes
+GH_ACCOUNT=	UniversityRadioYork
+GH_TAGNAME=	v0.3.1
```
Comment (LordAro) [inline ports/freebsd/Makefile]: Any way to deduplicate this with DISTVERSION above?

File: ports/freebsd/Makefile (hunk 4 above)
Comment (DinCahill) [inline ports/freebsd/Makefile]: I use `@ury.york.ac.uk` on GitHub, so I have to use it for my commits at least. I'll change it here. I'll have to look changing my GitHub too.

File: README.md (hunk 1 above)
Comment (DinCahill) [inline README.md]: ﻿ｆｍｌ

File: ports/freebsd/Makefile [hunk 6]
```diff
@@ -24,7 +24,7 @@ PLIST_FILES=	bin/playd man/man1/playd.1.gz
 
 USE_GITHUB=	yes
 GH_ACCOUNT=	UniversityRadioYork
-GH_TAGNAME=	v0.3.1
+GH_TAGNAME=	${DISTVERSION}
```
Comment (LordAro) [inline ports/freebsd/Makefile]: Now it's "0.3.1", instead of "v0.3.1". Intentional?

Comment (DinCahill) [inline ports/freebsd/Makefile]: Oops, fixed. I was rushing to go shopping!

File: .gitignore (hunk 2 above)
Comment (DinCahill) [inline .gitignore]: Fixed properly to `/Makefile` now.

**Pull request merged**